File: practica2_839756_842236/c++/instruccionJumpif.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionJumpif.h"

InstruccionJumpif::InstruccionJumpif(int _l)
{
    l = _l;
}

void InstruccionJumpif::ejecutar(stack<int> *s) {}

void InstruccionJumpif::ejecutar(stack<int> *s, int &counter)
{
    int x = s->top();
    s->pop();
    if (x >= 0)
    {
        counter = l;
    }
    else
    {
        counter++;
    }
}

void InstruccionJumpif::listar()
{
    cout << "jumpif " << l << endl;
}

File: practica2_839756_842236/java/InstruccionJumpif.java
import java.util.Stack;
import java.util.concurrent.atomic.AtomicInteger;

public class InstruccionJumpif extends Instruccion {
    private int l;

    public InstruccionJumpif(int _l) {
        l = _l;
    }

    @Override
    public void ejecutar(Stack<Integer> s) {
    }

    @Override
    public void ejecutar(Stack<Integer> s, AtomicInteger counter) {
        int x = s.pop();
        if (x >= 0) {
            counter.set(l);
        } else {
            counter.getAndIncrement();
        }
    }

    public void listar() {
        System.out.println("jumpif " + l);
    }
}
